packages/icons-react/src/IconFactory.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { ReactNode } from "react";
+import { variants } from "./icons/types";
+
+interface Props {
+    title: string;
+    description: string;
+    innerSvg: ReactNode;
+    variant?: variants;
+    viewBox?: string;
+    className?: string;
+}
+
+export const IconFactory: React.FC<Props> = ({
+    viewBox = "0 0 20 21",
+    variant = "small",
+    className = "",
+    title,
+    description,
+    innerSvg,
+}) => (
+    <svg
+        className={`jkl-icon jkl-icon--${variant} ${className}`}
```

Comment:
Jeg synes vi bør ha muligheten til å hente ut helt rå (React-)SVG av ikonene også, uten noen klasser etc. hengt på. Slik det er nå kan de for eksempel ikke enkelt brukes til å skalere sammen med tekst, siden man må overstyre størrelsen satt i `jkl-icon` og dermed styre med innlastingsrekkefølge av stilark etc.

Er nok mulig jeg misforsto dette da vi gikk gjennom det tidligere i dag, men jeg tolket `jkl-icon`-klassen mer som en nytteklasse for de vanligste tilfellene. Nå føles dette veldig som en fullverdig komponent, noe ikonene vel i utgangspunktet ikke skulle være? Blir jo selvfølgelig litt annerledes for de animerte, men tenkte det var her klassen og factory-komponenten kom inn..

Reply:
Tja, jeg gjorde det sånn fordi det hadde vært annoying å måtte styre størrelsen på alle ikon hver gang man bruker de, det åpner også for mye småfeil. Men om vi finner et annet system for sørge for konsekvent bruk, så er det fint. 

Men forsåvidt så endte denne factoryen opp med å bli litt rar, så kanskje likegreit å flytte alt ut i hver enkelt ikon, men hvordan skal vi styre konsekvente størrelser uten å sette klassen?

Reply:
> hvordan skal vi styre konsekvente størrelser uten å sette klassen?

Hvis ikonet skal stå sammen med tekst skal det vel alltid følge tekststørrelse, så man kan evt. ha `1em` som en default?

Jeg tror bare vi skal være forsiktige med å fastsette for mange egenskaper ved ikonene, nettopp siden de ikke egentlig eksisterer for seg selv men kun som del av andre komponenter. Hvis vi skal ha faste egenskaper tror jeg at vi e nødt til å trenke mer gjennom dem før vi legger dem ut: Selv om ting er placeholders har de en lei tendens til å bli godtatt som fakta veldig fort 😉